src/hooks/useOptionsClaim.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import { ethers } from "ethers";
+
+import { useConnection } from "./useConnection";
+import { useClaims, Claim } from "./useClaims";
+import { contracts } from "../config";
+
+const EMPTY: unique symbol = Symbol();
+
+type TxStatus = "idle" | "pending" | "resolved" | "rejected";
+type Tx = ethers.Transaction;
+type TxAction =
+  | {
+      type: "set error";
+      error: Error | null | string;
+    }
+  | {
+      type: "set transaction status";
+      txStatus: TxStatus;
+    }
+  | {
+      type: "set transaction";
+      tx: Tx | null;
+    };
+
+type TxState = {
+  error: Error | null | string;
+  txStatus: TxStatus;
+  tx: Tx | null;
+};
+
+function txReducer(state: TxState, action: TxAction) {
+  switch (action.type) {
+    case "set error": {
+      return {
+        ...state,
+        error: action.error,
+      };
+    }
+    case "set transaction status": {
+      return {
+        ...state,
+        txStatus: action.txStatus,
+      };
+    }
+    case "set transaction": {
+      return {
+        ...state,
+        tx: action.tx,
+      };
+    }
+    default: {
+      throw new Error(`Unsopported action type ${(action as any).type}`);
+    }
+  }
+}
+
+type TOptionsContext = TxState & {
+  claim: () => void;
+  claims?: Claim[];
+  isLoadingProof: boolean;
+  errorProof: Error | null;
+};
+const OptionsContext = React.createContext<typeof EMPTY | TOptionsContext>(
+  EMPTY
+);
+OptionsContext.displayName = "OptionsContext";
+
+export const OptionsProvider: React.FC = ({ children, ...delegated }) => {
+  const { address, network, signer } = useConnection();
+  const {
+    claims,
+    isLoading: isLoadingProof,
+    error: errorProof,
+    refetch,
+  } = useClaims();
+  const [txState, dispatch] = React.useReducer(txReducer, {
+    error: null,
+    txStatus: "idle",
+    tx: null,
+  });
+
+  // reset network state on address, or network change.
+  React.useEffect(() => {
+    dispatch({ type: "set transaction status", txStatus: "idle" });
+    dispatch({ type: "set error", error: null });
+    dispatch({ type: "set transaction", tx: null });
+  }, [address, network, signer]);
+
+  const claim = React.useCallback(
+    async (claimIdx = 0) => {
+      // check if we can actually claim options for this claimIdx.
+      if (!(claims && claims.length > claimIdx && network?.chainId && signer)) {
+        if (!(claims && claims.length > claimIdx)) {
+          dispatch({
+            type: "set error",
+            error: "There are no options to claim for this address.",
```

Comment:
nit: should we have an error message for the `network?.chainId && signer` related issues? It seems like we're just returning and doing nothing in that case...